### PR TITLE
One Configuration to rule them all!

### PR DESCRIPTION
**Pull request by tfenne**
Added further nesting of config in the initialize() method.
Merged Configuration and DagrConfig into one cohesive Object (I think).
Cleaned up imports of anything referencing DagrConfig.

### CODE REVIEW

File: core/src/main/scala/dagr/core/config/Configuration.scala [hunk 1]
```diff
@@ -27,32 +27,79 @@ import java.io.File
...
+  /** Implement the abstract method from BaseConfiguration to return the actual config instance. */
+  override private[config] def config = _config
+
+  // Keys for configuration values used in dagr core
+  object Keys {
+    val CommandLineName = "dagr.command-line-name"
+    val ScriptDirectory = "dagr.script-directory"
+    val LogDirectory = "dagr.log-directory"
+    val SystemCores = "dagr.system-cores"
+    val SystemMemory = "dagr.system-memory"
```
Comment (nh13) [inline core/src/main/scala/dagr/core/config/Configuration.scala]: what happened to the `=` alignment?  ;)

Comment (tfenne) [inline core/src/main/scala/dagr/core/config/Configuration.scala]: I didn't even notice - freaking IntelliJ and it's need to mess with my whitespace on paste.

File: core/src/main/scala/dagr/core/config/Configuration.scala [hunk 2]
```diff
@@ -165,3 +212,8 @@ trait Configuration extends LazyLogging {
     */
   private def systemPath : Seq[Path] = config.getString("dagr.path").split(File.pathSeparatorChar).view.map(Paths get _)
 }
+
+trait Configuration extends ConfigurationLike {
+  /** Grabs a reference to the global configuration at creation time. */
+  override private[config] val config: Config = Configuration.config
+}
```
Comment (nh13) [inline core/src/main/scala/dagr/core/config/Configuration.scala:219]: nl

**Comment by nh13** [comment]
Looks great, I really like that we can have `application.conf` packaged in and it will load into the config.  Kudos.

**Pull request merged**